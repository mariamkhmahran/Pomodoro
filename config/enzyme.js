import { configure } from "enzyme";
import Adapter from "enzyme-react-adapter-future";

configure({ adapter: new Adapter() });
