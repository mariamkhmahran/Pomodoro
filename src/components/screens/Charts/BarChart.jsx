import React from "react";
import { Bar } from "react-chartjs-2";
import * as GeneralState from "../../../logic/GeneralState";

var weekday = new Array(7);
weekday[0] = "Sunday";
weekday[1] = "Monday";
weekday[2] = "Tuesday";
weekday[3] = "Wednesday";
weekday[4] = "Thursday";
weekday[5] = "Friday";
weekday[6] = "Saturday";

var chartOptions = {
  maintainAspectRatio: false,
  scales: {
    xAxes: [
      {
        gridLines: {
          display: false,
          color: "white"
        },
        ticks: {
          beginAtZero: true,
          fontColor: "white"
        }
      }
    ],
    yAxes: [
      {
        gridLines: {
          display: false,
          color: "white"
        },
        ticks: {
          beginAtZero: true,
          display: false
        }
      }
    ]
  }
};

export default class BarChart extends React.Component {
  state = {
    data: {
      labels: [],
      datasets: [
        {
          label: "Cycles",
          backgroundColor: "#F36969",
          borderColor: "#F36969",
          data: []
        },
        {
          label: "Tasks",
          backgroundColor: "#8EB7EE",
          borderColor: "#8EB7EE",
          data: []
        }
      ]
    }
  };

  async componentDidMount() {
    let newData = await GeneralState.getChartData();
    let data = this.state.data;
    data.labels = newData.cycles.map(d => weekday[d.day]);
    data.datasets[0].data = newData.cycles.map(d => d.value);
    data.datasets[1].data = newData.tasks.map(d => d.value);
    this.setState({ data });

    this.props.onDidMount && this.props.onDidMount(this);
  }

  render() {
    return (
      <div
        style={{
          width: "90%",
          height: "100%"
        }}>
        <Bar
          id="chart"
          data={this.state.data}
          options={chartOptions}
          height={500}
          width={700}
          onChange={this.changeHandler}
        />
      </div>
    );
  }
}
