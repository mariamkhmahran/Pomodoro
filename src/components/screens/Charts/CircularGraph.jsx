import React from "react";

export default class CircularGraph extends React.Component {
  componentDidMount() {
    console.log(this.props);
    let { x, y, id, percentage } = this.props;
    this.setState({ percentage: percentage });

    let arc = percentage * 6.3 + -1.6;
    var c = document.getElementById(id);
    var ctx = c.getContext("2d");
    ctx.beginPath();
    ctx.arc(x, y, 60, arc, 1.5 * Math.PI);
    ctx.lineWidth = 10;
    ctx.strokeStyle = "white";
    ctx.stroke();
  }

  render() {
    let { id } = this.props;
    return <canvas id={id} width="200" height="200" style={{ height: "250px", width: "250px" }} />;
  }
}
