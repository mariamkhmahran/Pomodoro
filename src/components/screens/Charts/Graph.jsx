var React = require("react");
var CanvasJSReact = require("./canvasjs.react");
var CanvasJSChart = CanvasJSReact.CanvasJSChart;

export default class Graph extends React.Component {
  toggleDataSeries = e => {
    if (e.dataSeries.visible === "undefined" || e.dataSeries.visible) {
      e.dataSeries.visible = false;
    } else {
      e.dataSeries.visible = true;
    }
    this.chart.render();
  };

  render() {
    const options = {
      theme: "light1",
      animationEnabled: true,
      axisX: {
        title: ""
      },
      axisY: {
        title: "",
        titleFontColor: "transparent",
        lineColor: "white",
        labelFontColor: "transparent",
        tickColor: "transparent",
        includeZero: true
      },
      axisY2: {
        title: "",
        titleFontColor: "transparent",
        lineColor: "transparent",
        labelFontColor: "transparent",
        tickColor: "transparent",
        includeZero: true
      },
      toolTip: {
        shared: true
      },
      legend: {
        cursor: "pointer",
        itemclick: this.toggleDataSeries
      },
      data: [
        {
          type: "spline",
          name: "tasks",
          showInLegend: false,
          xValueFormatString: "M YYYY",
          yValueFormatString: "#,##0 Units",
          dataPoints: [
            { x: new Date(2018, 1, 1), y: 2 },
            { x: new Date(2018, 1, 2), y: 3 },
            { x: new Date(2018, 1, 3), y: 1 },
            { x: new Date(2018, 1, 4), y: 5 },
            { x: new Date(2018, 1, 5), y: 5 },
            { x: new Date(2018, 1, 6), y: 3 },
            { x: new Date(2018, 1, 7), y: 5 },
            { x: new Date(2018, 1, 8), y: 1 },
            { x: new Date(2018, 1, 9), y: 4 },
            { x: new Date(2018, 1, 10), y: 2 },
            { x: new Date(2018, 1, 11), y: 2 },
            { x: new Date(2018, 1, 12), y: 2 }
          ]
        },
        {
          type: "spline",
          name: "cycles",
          axisYType: "secondary",
          showInLegend: false,
          xValueFormatString: "MM YYYY",
          yValueFormatString: "$#,##0.#",
          dataPoints: [
            { x: new Date(2018, 1, 1), y: 12 },
            { x: new Date(2018, 1, 2), y: 10 },
            { x: new Date(2018, 1, 3), y: 10 },
            { x: new Date(2018, 1, 4), y: 11 },
            { x: new Date(2018, 1, 5), y: 10 },
            { x: new Date(2018, 1, 6), y: 8 },
            { x: new Date(2018, 1, 7), y: 8 },
            { x: new Date(2018, 1, 8), y: 9 },
            { x: new Date(2018, 1, 9), y: 10 },
            { x: new Date(2018, 1, 10), y: 11 },
            { x: new Date(2018, 1, 11), y: 12 },
            { x: new Date(2018, 1, 12), y: 12 }
          ]
        }
      ]
    };

    return (
      <div>
        <CanvasJSChart options={options} onRef={ref => (this.chart = ref)} />
      </div>
    );
  }
}
