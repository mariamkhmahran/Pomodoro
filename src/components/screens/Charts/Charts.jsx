import React from "react";
import BarChart from "./BarChart";

export default class Charts extends React.Component {
  render() {
    return (
      <div className="router_output">
        <div className="subHeader">
          <span className="subtitle">Charts</span>
        </div>
        <div className="Main charts">
          <div>
            <div
              className="transparent_box"
              style={{ height: "450px", width: "100%" }}>
              <BarChart />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
