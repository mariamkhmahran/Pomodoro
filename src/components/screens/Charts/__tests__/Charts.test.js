import React from "react";
import { shallow } from "enzyme";
import Charts from "src/components/screens/Charts";

test("Charts.render", () => {
  const wrapper = shallow(<Charts />, { lifecycleExperimental: true });

  expect(wrapper).toMatchSnapshot();
});
