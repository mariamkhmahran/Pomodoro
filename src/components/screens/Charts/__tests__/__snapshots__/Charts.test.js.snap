// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts.render 1`] = `
<div
  className="router_output"
>
  <div
    className="subHeader"
  >
    <span
      className="subtitle"
    >
      Charts
    </span>
  </div>
  <div
    className="Main charts"
  >
    <div>
      <div
        className="transparent_box"
        style={
          Object {
            "height": "450px",
            "width": "100%",
          }
        }
      >
        <BarChart />
      </div>
    </div>
  </div>
</div>
`;
